### PR TITLE
Add labels to render

### LINKED ISSUE #3482: Support setting labels in `skaffold render`
when deploying with `kubectl` or `kustomize`, currently the manifests are rendered using `renderManifests`, and then labels are set in the manifests after they're rendered. this same function is used for `skaffold render`, meaning that labels are not set in the final rendered manifests and don't match up to the manifests that are actually sent through to be deployed.

### PR DESCRIPTION
**Pull request by arminbuerkle**
Fixes #3482.

**Description**
Adds the `--label` option to `skaffold render`.

@nkubala didn't have much time the last weeks, but here is the follow up of https://github.com/GoogleContainerTools/skaffold/pull/3110#issuecomment-563960181

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [x] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**
Add `--label` option to `skaffold render`

### CODE REVIEW

File: pkg/skaffold/deploy/kubectl_test.go [hunk 1]
```diff
@@ -491,12 +493,26 @@ func TestKubectlRender(t *testing.T) {
 					Tag:       "gcr.io/k8s-skaffold/skaffold:test",
 				},
 			},
+			labels: []Labeller{},
 			input: `apiVersion: v1
 kind: Pod
+metadata:
```
Comment (tejal29) [inline pkg/skaffold/deploy/kubectl_test.go]: was there a reason for adding this in input?
```
  metadata
     namespace: default
```

**Comment by tejal29** [comment]
@arminbuerkle i rebased your branch on latest master and added another test

```
+               {
+                       description: "single kustomization with user labels",
+                       builds: []build.Artifact{
+                               {
+                                       ImageName: "gcr.io/project/image1",
+                                       Tag:       "gcr.io/project/image1:tag1",
+                               },
+                               {
+                                       ImageName: "gcr.io/project/image2",
+                                       Tag:       "gcr.io/project/image2:tag2",
+                               },
+                       },
+                       labels: []Labeller{
+                               &testLabels{
+                                       labels: map[string]string{
+                                               "user/label": "test",
+                                       }},
+                       },
+                       kustomizations: []kustomizationCall{
+                               {
+                                       folder: ".",
+                                       buildResult: `apiVersion: v1
+kind: Pod
+metadata:
+  namespace: default
+spec:
+  containers:
+  - image: gcr.io/project/image1
+    name: image1
+  - image: gcr.io/project/image2
+    name: image2
+`,
+                               },
+                       },
+                       expected: `apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    skaffold.dev/deployer: kustomize
+    user/label: test
+  namespace: default

```

If you give me permission to push to your branch, i can update this PR and merge it.

File: pkg/skaffold/deploy/kubectl_test.go (hunk 1 above)
Comment (arminbuerkle) [reply pkg/skaffold/deploy/kubectl_test.go]: Yes! If `metadata` doesn't exist, the default labeller doesn't execute, therefore not adding any labels.

Without it `skaffold.dev/deployer: kubectl` won't be added.

**Comment by arminbuerkle** [comment]
> If you give me permission to push to your branch, i can update this PR and merge it.

@tejal29 i added you as a collaborator to our repository. For some reason i don't have the `allow maintainers to edit` checkbox in this PR?

**Comment by tejal29** [comment]
@arminbuerkle i got the request! Thanks!

**Pull request merged**